pkg/cloud/aws/provider.go

```diff
@@ -1307,9 +1321,14 @@ func (aws *AWS) createNode(terms *AWSProductTerms, usageType string, k models.Ke
 		}, meta, nil
 	} else if aws.isPreemptible(key) { // Preemptible but we don't have any data in the pricing report.
 		log.DedupedWarningf(5, "Node %s marked preemptible but we have no data in spot feed", k.ID())
+		// Throw error if public price is not found
+		if !publicPricingFound {
+			log.Errorf("Could not fetch public pricing data for spot instance \"%s\"", k.ID())
+			return nil, meta, fmt.Errorf("Could not fetch data for \"%s\"", k.ID())
+		}
 		return &models.Node{
+			Cost:         cost,
 			VCPU:         terms.VCpu,
-			VCPUCost:     aws.BaseSpotCPUPrice,
 			RAM:          terms.Memory,
 			GPU:          terms.GPU,
 			Storage:      terms.Storage,
```

Comment:
In the case that a Spot Data Feed isn't available, I'm unsure if we should be creating a dependency on the AWS OnDemand Pricing Sheet?

Reply:
Or maybe in the case that `!publicPricingFound`, we can set use defaults here instead of throwing an error?

Reply:
Thanks for the review @thomasvn ! I have made the change to return defaults when public price is not available.

I have made one change while returning defaults though: I am returning both vCPU and RAM default price instead of just vCPU price like earlier. If you want me to change that, please let me know. Thanks!

Reply:
Thanks for this change. I'm in agreement here!